plugins/ltac/tacsubst.ml

```diff
@@ -171,17 +171,20 @@ let rec subst_atomic subst (t:glob_atomic_tactic_expr) = match t with
   | TacInversion (InversionUsing (c,cl),hyp) ->
       TacInversion (InversionUsing (subst_glob_constr subst c,cl),hyp)
 
-and subst_tactic subst (t:glob_tactic_expr) = match t with
-  | TacAtom { CAst.v=t } -> TacAtom (CAst.make @@ subst_atomic subst t)
+and subst_tactic subst (t:glob_tactic_expr) =
+  let (loc, t) = CAst.(t.loc, t.v) in
+  let t' = match t with
```

Comment:
Seems like this should be
```suggestion
and subst_tactic subst = CAst.map (function
```
(+ closing parens at the end)

Reply:
The `let` is needed to preserve the value of `loc`.  If I change this to

```ocaml
and subst_tactic subst (t:glob_tactic_expr) =
  let (loc, t) = CAst.(t.loc, t.v) in
  CAst.map ?loc (match t with
```

and tweak the end of the function, I get a compile error:

```
File "plugins/ltac/tacsubst.ml", line 176, characters 12-15:
Error: The function applied to this argument has type 'a CAst.t -> 'b CAst.t
This argument cannot be applied with label ?loc
```

Reply:
That's not what I wrote.
map preserves loc, that's what it's for, and it's why it doesn't take a ?loc argument

Reply:
I did forget to mention that using `map` the `cast.make` at the end also gets removed though

Reply:
Got it.  After editing `CAst.make` so much lately, it so easty to misread `CAst.map` :-)

Reply:
Indeed `CAst.map` is not only used for convenience, but must be also be used for correctness, as other attributes may appear in the CAst node in the future.